integration-tests/contract-staking.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { Protocols } from "@taquito/taquito";
+import { CONFIGS, isSandbox, sleep } from "./config";
+
+CONFIGS().forEach(({ lib, rpc, setup, protocol }) => {
+    const Tezos = lib;
+    describe(`Test staking through contract API using: ${rpc}`, () => {
+      const flextesaOxford = isSandbox({rpc}) && protocol === Protocols.Proxford ? test : test.skip;
+      beforeEach(async (done) => {
+          await setup(true)
+          done()
+        });
+        flextesaOxford('Should be able to stake', async (done) => {
+          const op = await Tezos.contract.stake({
+            amount: 100,
+          });
+          await op.confirmation()
+          expect(op.hash).toBeDefined();
+          
+          const constants = await Tezos.rpc.getConstants();
+          await sleep(((constants.preserved_cycles + 2) * constants.blocks_per_cycle * (constants.minimal_block_delay!.toNumber())) * 1000);
```

Comment:
from my experiment seems to be able to unstake immediately after stake might not need to sleep (the wait between stake and unstake is to get attestationRight/bakingRight)

Reply:
Without the sleeps, I would get some errors in some runs. I can investigate that further.

Reply:
best to figure out why the errors happen without the sleep. Relying on sleeps here might mean users would need to introduce sleeps as well(?)

Reply:
is this finalized?